### PR TITLE
chore(workflow): split ci workflow in multiple workflows & fix regex

### PR DESCRIPTION
**Pull request by carlocorradini**

### CODE REVIEW

File: .github/workflows/check.yml [hunk 1]
```diff
@@ -1,11 +1,11 @@
-name: ci
+name: check
 
 on:
   push:
     branches:
       - master
     tags:
-      - v*
+      - "^v*"
```
Comment (MichalLytek) [inline .github/workflows/check.yml:8]: 1) Is this necessary? I'm not publishing any other tags, especially tags starting with `v` like `vercel` or something
2) Is this tested on your fork? Last time we've merged not working stuff, here I'm afraid of `^` not being glob compliant

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+      - name: Build Changelog
+        id: changelog
+        uses: mikepenz/release-changelog-builder-action@v4
+        with:
+          configuration: "./.github/configs/changelog.json"
+          owner: MichalLytek
+          repo: type-graphql
+          failOnError: true
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```
Comment (MichalLytek) [inline .github/workflows/release.yml:75]: Can you show me how it works on your fork? The changelog for beta 2 release.

File: .github/workflows/check.yml (hunk 1 above)
Comment (carlocorradini) [reply .github/workflows/check.yml]: @MichalLytek Right! Fixed

File: .github/workflows/release.yml (hunk 2 above)
Comment (carlocorradini) [reply .github/workflows/release.yml]: Tried and it should work. I really don't know why...

**Review by MichalLytek — APPROVED** [review]

**Pull request merged**